eslint.config.js

```diff
@@ -27,6 +28,7 @@ module.exports = {
     // This updates the rule below to throw a warning rather than an error
     // when using eslint-plugin-react 7.12.2 or above.
     'react/jsx-wrap-multilines': 'warn',
+    'react-hooks/rules-of-hooks': 'error',
```

Comment:
Will this require a major version bump?

Reply:
I wouldn't think so, what are your thoughts?

Reply:
I would say this is technically non-passive because it will break teams that have used hooks but violated these rules, but it's so close to the release of hooks that teams are likely just starting development and react strongly encourages we enforce these rules.

Reply:
If we are worried about passivity, we could set this as 'warn' for now?

Reply:
I talked with the other maintainers on this and the plan is to treat this update as a passive release. I can understand the perspective of this being non-passive for people who used hooks in the react alpha versions but used them in ways the react team recommended against. I think it's a small edge case and we'd be fine considering this a passive release.